src/Driver/AbstractDB2Driver.php

```diff
@@ -49,4 +53,41 @@ public function getExceptionConverter(): ExceptionConverterInterface
     {
         return new ExceptionConverter();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createDatabasePlatformForVersion($version)
+    {
+        if (version_compare($this->getVersionNumber($version), '11.1', '>=')) {
+            return new DB2111Platform();
+        }
+
+        return $this->getDatabasePlatform();
```

Comment:
Should we already trigger a deprecation for this case? [According to my Google research](https://www.ibm.com/support/pages/db2-distributed-end-support-eos-dates), extended support for DB2 10.5 ends in less than three weeks and all other releases below 11.1 have already reached their end of life. We could drop support for DB2 < 11.1 in 4.0.

Reply:
Looking at the implementation of `doModifyLimitQuery()` on the 4.0.x branch, I guess this decision has already been made. In that case, please trigger a deprecation in case we instantiate the old platform class. Also, the `DB2111Platform` class should be flagged as `@deprecated` because we will remove it in 4.0.x.

Reply:
The suggested deprecation messages were added.